### PR TITLE
Fail QueryWorkflow if last workflow task failed

### PR DESCRIPTION
**Pull request by samanbarghi**
**What changed?**
Fail fast for query if the last workflow task has failed.

**Why?**
While workflow task is in failing state (due to non-deterministic error etc), the query to those workflow will fail anyway, and deliver query to the workflow will only waste resources to load history. 
We should fail fast for query if we know the workflow task would fail.

**How did you test it?**
New functional test.

**Potential risks**
No risks.

**Is hotfix candidate?**
No.

### CODE REVIEW

File: tests/query_workflow_test.go [hunk 1]
```diff
@@ -257,3 +258,41 @@ func (s *clientIntegrationSuite) TestQueryWorkflow_QueryBeforeStart() {
...
+	workflowRun, err := s.sdkClient.ExecuteWorkflow(ctx, workflowOptions, workflowFn)
+	if err != nil {
+		s.Logger.Fatal("Start workflow failed with err", tag.Error(err))
+	}
+
+	s.NotNil(workflowRun)
+	s.True(workflowRun.GetRunID() != "")
+
+	// wait for workflow to fail
+	time.Sleep(time.Second * 5)
```
Comment (yycptt) [inline tests/query_workflow_test.go:295]: nit: do we have to wait for 5 sec? seems a bit long. I think workflow task attempt will be larger than 1 as soon as the first workflow task is failed.

File: tests/query_workflow_test.go [hunk 2]
```diff
@@ -257,3 +258,41 @@ func (s *clientIntegrationSuite) TestQueryWorkflow_QueryBeforeStart() {
...
+
+func (s *clientIntegrationSuite) TestQueryWorkflow_QueryFailedWorkflowTask() {
+
+	workflowFn := func(ctx workflow.Context) (string, error) {
+		err := workflow.SetQueryHandler(ctx, "test", func() (string, error) {
+			return "", nil
+		})
+
+		if err != nil {
+			s.Logger.Fatal("SetQueryHandler failed: " + err.Error())
```
Comment (yiminc) [inline tests/query_workflow_test.go:270]: I think Fatel may exit the process or fail the test?

File: service/history/api/queryworkflow/api.go [hunk 3]
```diff
@@ -115,6 +115,12 @@ func Invoke(
 		return nil, consts.ErrWorkflowTaskNotScheduled
 	}
 
+	if mutableState.GetExecutionInfo().WorkflowTaskAttempt > 1 {
+		// while workflow task is failing, the query to that workflow will also fail. Failing fast here to prevent wasting
+		// resources to load history for a query that will fail.
+		return nil, serviceerror.NewFailedPrecondition("Query has failed due to a failing workflow task")
```
Comment (yiminc) [inline service/history/api/queryworkflow/api.go:121]: Cannot query workflow due to Workflow Task in failed state.

**Review by yycptt — APPROVED** [review]
Need this fix for 1.20.1 patch release.

**Pull request merged**